File: 백준/Silver/1912. 연속합/연속합.py
import sys
input = sys.stdin.readline

n = int(input())
arr = list(map(int,input().split()))
s = [0] * n
s[0] = arr[0]
for i in range(1, n):
    s[i] = max(s[i - 1] + arr[i] , arr[i])

#print(s)
print(max(s))

File: 백준/Silver/1912. 연속합/연속합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        int n = Integer.parseInt(br.readLine());

        int[] arr = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int[] prefix = new int[n];
        prefix[0] = arr[0];
        int ans = prefix[0];

        for (int i = 1; i < n; i++) {
            prefix[i] = Math.max(prefix[i - 1] + arr[i], arr[i]);
            ans = Math.max(prefix[i], ans);
        }
        System.out.println(ans);
    }
}
